### ISSUE TITLE
Missing metadata

### ISSUE THREAD

**Issue by joe-pitt** [openghg/openghg#1194]
### What is your issue?

Sometime since v0.10.1 there has been a change resulting in many metadata fields no longer being present when icos data is loaded into a store using retrieve. I can't find any big changes to the retrieve function itself, my best guess is that it has something to do with this commit: https://github.com/openghg/openghg/commit/6323633de44ec2f05423fda41ea48a041b06dd67. I could be completely wrong here though as this stuff is beyond me.

The missing fields are: `inlet, inlet_height_magl, instrument, sampling_period, calibration_scale, data_owner, data_owner_email, station_height_masl`

This is a fairly urgent one I'm afraid as I'm trying to address Issue #1190 in order to regenerate the CH4 obs files for PARIS. I could roll back to v0.10.1 and apply the #1190 fix, but things could easily get in a mess that way...

**Comment by joe-pitt** [comment]
To see what I mean, you can run the following under v0.10.1 and on the current devel:
```
from openghg.retrieve.icos import retrieve_atmospheric
from icoscp_core.icos import bootstrap
from icoscp import cpauth

# New auth method with token
# Need to login to ICOS account every day to get new token and paste it here
cookie_token = ""
meta, data = bootstrap.fromCookieToken(cookie_token)
cpauth.init_by(data.auth)

cp_data = retrieve_atmospheric(site="BIK",
                               species = "CO2",
                               force_retrieval=True,
                               data_level=2,
                               dataset_source="European ObsPack",
                               update_mismatch="from_source",
                               store="user")
cp_data[0].metadata
```

You have to get a cookie token first by logging into your icoscp account (due to the new authorisation step)

**Comment by rt17603** [comment]
> cp_data = retrieve_atmospheric(site="BIK",
>                                species = "CO2",
>                                force_retrieval=True,
>                                data_level=2,
>                                dataset_source="European ObsPack",
>                                update_mismatch="from_source",
>                                store="user")
> cp_data[0].metadata

I'm having trouble reproducing this as I can't seem to get the `retrieve_atmospheric` function to run at all at the moment (even following the authorisation steps) - can I suggest reverting to 0.10.1 for now. I'm not sure I'd be able to resolve this today and I'm then on annual leave until after Christmas.

@brendan-m-murphy - were you able to take a look at this or were you similarly away soon?

> I can't find any big changes to the retrieve function itself, my best guess is that it has something to do with this commit: https://github.com/openghg/openghg/commit/6323633de44ec2f05423fda41ea48a041b06dd67
FYI, I'm not sure it is related to the commit you pointed to as that *should* just be a name change and so should not impact anything.

**Comment by joe-pitt** [comment]
Ah OK I've done some testing and it is when I checkout this commit that I lose the additional metadata: https://github.com/openghg/openghg/commit/5eea4a4eb88a8581e44699cc3dd9ff31a9874951. Does this have something to do with how sync_surface_metadata is being called?

**Comment by joe-pitt** [comment]
I've done some digging - I can't claim to fully understand the details of this, but I think the issue is that sync_surface_metadata is now in the read_file method of ObsSurface. This is used within standardise but not the icos retrieve function. I'm going to try reversing PR #1138 for now and see if that fixes things.

**Comment by joe-pitt** [comment]
Yeah that is now back working again - I'll make a PR for this and you folks can edit as needed

**Comment by SutarPrasad** [comment]
I have raised a new PR without shifting the sync_surface_metadata call to the previous location.
Below is the data obtained after the new changes, Can you please confirm if everything is as expected @joe-pitt
In [35]: cp_data = retrieve_atmospheric(
    ...:         site="JFJ",
    ...:         species="ch4",
    ...:         force_retrieval=True,
    ...:         update_mismatch="metadata",
    ...:         store="user",
    ...:     )
2025-01-06T12:32:54 INFO     INFO:openghg.retrieve:Retrieving https://meta.icos-cp.eu/objects/9BmAUqYgyNKW2hWP-Kq0X5sI...                           _retrieve.py:318
2025-01-06T12:32:56 INFO     INFO:openghg.retrieve:Retrieving https://meta.icos-cp.eu/objects/F5LbbXnWZ9sW5ODzzMCOALJA...                           _retrieve.py:318
2025-01-06T12:32:57 INFO     INFO:openghg.retrieve:Retrieving https://meta.icos-cp.eu/objects/_56CZ8aqDBlVG275JGOK_nd3...                           _retrieve.py:318
2025-01-06T12:32:59 WARNING  WARNING:openghg.standardise.metadata:Metadata mismatch for 'station_long_name', metadata: Jungfraujoch, Switzerland -  _metadata.py:141
                             attributes: JFJ                                                                                                                        
                             Updating attributes to use metadata value: station_long_name = Jungfraujoch, Switzerland                                               
                    WARNING  WARNING:openghg.standardise.metadata:Metadata mismatch for 'station_long_name', metadata: Jungfraujoch, Switzerland -  _metadata.py:141
                             attributes: JFJ                                                                                                                        
                             Updating attributes to use metadata value: station_long_name = Jungfraujoch, Switzerland                                               
                    WARNING  WARNING:openghg.standardise.metadata:Metadata mismatch for 'station_long_name', metadata: Jungfraujoch, Switzerland -  _metadata.py:141
                             attributes: JFJ                                                                                                                        
                             Updating attributes to use metadata value: station_long_name = Jungfraujoch, Switzerland                                               

In [36]: cp_data[0].metadata
Out[36]: 
{'station_long_name': 'jungfraujoch, switzerland',
 'station_latitude': 46.5475,
 'station_longitude': 7.9851,
 'species': 'ch4',
 'network': 'icos',
 'data_type': 'surface',
 'data_source': 'icoscp',
 'source_format': 'icos',
 'data_level': '2',
 'site': 'jfj',
 'inlet': '13.9m',
 'inlet_height_magl': '13.9',
 'instrument': 'multiple',
 'sampling_period': 'not_set',
 'calibration_scale': 'unknown',
 'data_owner': 'yves-alain roulet',
 'data_owner_email': 'yves-alain.roulet@meteoswiss.ch',
 'station_height_masl': 3571.8,
 'dataset_source': 'ICOS'}
 
 I have checked for all the missing metadata mentioned in this issue and it seems to be getting fetched correctly in the new changes.
 
PR linked: https://github.com/openghg/openghg/pull/1197
